lib/dor/models/identifiable.rb

```diff
@@ -224,6 +224,19 @@ def remove_druid_prefix(druid=id)
       result.nil? ? druid : result[0]  # if no matches, return the string passed in, otherwise return the match 
     end
 
+    # Override ActiveFedora::Core#adapt_to_cmodel (used with associations, among other places) to
+    # preferentially use the objectType asserted in the identityMetadata.
+    def adapt_to_cmodel
+      object_type = identityMetadata.objectType.first
+      object_class = Dor.registered_classes[object_type]
+
+      if object_class
```

Comment:
for my clarification, doesn't the `registered_classes[object_type]` call throw `NameError` if the `object_type` is missing? if so, I don't think this `if` condition is ever false. that is, when would `super` ever get called?